docs/DeveloperGuide.md

```diff
@@ -4,26 +4,167 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
```

Comment:
Consider adding an overall class diagram and introduce the major components so that it would not very confusing below when components are named suddenly.